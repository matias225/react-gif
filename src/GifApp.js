import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid';

const GifApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);
    
    return (
        <>
            <h2>GifApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>

            <ol>
                { 
                    categories.map( category => 
                        <GifGrid 
                            key={ category }
                            category={ category } 
                        />
                    ) 
                }
            </ol>
        </>
    )
}

export default GifApp