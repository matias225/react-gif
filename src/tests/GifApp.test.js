import React from "react";
import GifApp from "../GifApp";
import { shallow } from "enzyme";

describe('Pruebas en el GifApp', () => {
    
    const wrapper = shallow( <GifApp /> );

    test('Debe cargar correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['One Piece', 'Dragon Ball Z'];
        const wrapper = shallow( <GifApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });

});