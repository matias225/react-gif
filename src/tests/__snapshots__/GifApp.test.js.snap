// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifApp Debe cargar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el GifApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;
